simplekv/net/azurestore.py

```diff
@@ -142,16 +142,20 @@ def __init__(self, block_blob_service, container_name, key):
 
     def tell(self):
         """Returns he current offset as int. Always >= 0."""
+        if self.closed:
+            raise ValueError("I/O operation on closed file")
         return self.pos
 
     def read(self, size=-1):
         """Returns 'size' amount of bytes or less if there is no more data.
         If no size is given all data is returned. size can be >= 0."""
+        if self.closed:
+            raise ValueError("I/O operation on closed file")
         with map_azure_exceptions(key=self.key):
             if size < 0:
                 size = self.size - self.pos
 
-            end = min(self.pos + size - 1, self.size)
+            end = min(self.pos + size - 1, self.size - 1)
```

Comment:
Can we have a test which fails on current master and succeeds with the change?

Reply:
I'm not sure. The azure API does not care if you request more than the actual size. It just gives you the all the bytes it can. So in my opinion the current implementation should have also worked fine. So the new implementation is only more correct, but behaves same as the previous one.

```
block_blob_service = BlockBlobService(account_name=ACCOUNT_NAME, account_key=ACCOUNT_KEY)
size = b.properties.content_length
b = block_blob_service.get_blob_properties("parquet", "test")
b = block_blob_service.get_blob_to_bytes("parquet", "test", start_range=size-1, end_range=size)
b.content
```